packages/kobotoolbox/src/Adaptor.js

```diff
@@ -52,52 +44,76 @@ export function execute(...operations) {
 export function getForms() {
   return async state => {
     const url = `/assets/?asset_type=survey`;
-
     const response = await util.request(state, 'GET', url, {});
 
-    console.log('✓', response.body.results.length, 'forms fetched.');
+    console.log('✓', response.body.results?.length, 'forms fetched.');
+
     return util.prepareNextState(state, response);
   };
 }
 
 /**
  * Get submissions for a specific form. Calls `/api/v2/assets/<formId>/data/`.
  * @example <caption>Get all submissions for a specific form</caption>
- * getSubmissions('aXecHjmbATuF6iGFmvBLBX');
+ * getSubmissions('aXecHjmbATuF6iGFmvBLBX', {limit: Infinity});
  * @example <caption>Get form submissions with a query</caption>
- * getSubmissions('aXecHjmbATuF6iGFmvBLBX', { query: { _submission_time:{ $gte: "2022-06-12T21:54:20" } } });
+ * getSubmissions('aXecHjmbATuF6iGFmvBLBX', { query: { _submission_time:{ $gte: "2025-03-12T21:54:20" } } });
  * @function
  * @public
  * @param {string} formId - Form Id to get the specific submissions
  * @param {object} [options={}] - Optional query params for the request
+ * @param {number} [options.limit] - Limit the number of submissions to fetch
```

Comment:
What do you think about adding something like: `restricts query to a single request. Use max to allow multiple requests`

Reply:
I don't think this is true, Because it depends on `pageSize`

Reply:
If limit is set, pageSize and max are ignored
```
if (keysToRemove.length) {
        console.warn(
          `Warning: ignoring option [${keysToRemove.join(
            ','
          )}] as "limit" is set`
        );
        delete resolvedOptions.max;
        delete resolvedOptions.pageSize;
      }
```